gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +720,684 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+                not the center of Y Cartesian system (which is latitude_of_origin)
+                But FITS WCS only supports projections on the sphere
+                we assume here that the local radius is the one computed at the projection center
+                */
+                oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                oSRS.SetProjParm(SRS_PP_STANDARD_PARALLEL_1, crval2);
+                oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+                oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+
+            /* Lambert Azimuthal Equal Area / ZEA projection */
+            } else if( strcmp(projstr,"ZEA" ) == 0 ) {
+                strncpy(projName, "Lambert_Azimuthal_Equal_Area_", 30);
+                oSRS.SetProjection(SRS_PT_LAMBERT_AZIMUTHAL_EQUAL_AREA);
+                oSRS.SetProjParm(SRS_PP_LONGITUDE_OF_CENTER, crval1);
+                oSRS.SetProjParm(SRS_PP_LATITUDE_OF_CENTER, crval2);
+
+            /* Lambert Conformal Conic 1SP / COO projection */
+            } else if( strcmp(projstr,"COO" ) == 0 ) {
+                strncpy(projName, "Lambert_Conformal_Conic_1SP_", 29);
+                oSRS.SetProjection(SRS_PT_LAMBERT_CONFORMAL_CONIC_1SP);
+                oSRS.SetProjParm(SRS_PP_LONGITUDE_OF_CENTER, crval1);
+                /*
+            #clat = self.__header['XXXXX']
+            #srs.SetProjParm('latitude_of_center',clat)
+            scale = self.__header['XXXXX']
+            if scale is not None:
+                srs.SetProjParm('scale_factor',scale)
+            else: #set default of 1.0
+                srs.SetProjParm('scale_factor',1.0)
+                */
+
+            /* Orthographic / SIN projection */
+            } else if( strcmp(projstr,"SIN" ) == 0 ) {
+                strncpy(projName, "Orthographic_", 14);
+                oSRS.SetProjection(SRS_PT_ORTHOGRAPHIC);
+                oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                /*
+            #olat = self.__header['XXXXX']
+            #srs.SetProjParm('latitude_of_origin',olat)
+                */
+
+            /* Point Perspective / AZP projection */
+            } else if( strcmp(projstr,"AZP" ) == 0 ) {
+                strncpy(projName, "perspective_point_height_", 26);
+                oSRS.SetProjection(SRS_PP_PERSPECTIVE_POINT_HEIGHT);
+            /* # appears to need height... maybe center lon/lat */
+
+            /* Polar Stereographic / STG projection */
+            } else if( strcmp(projstr,"STG" ) == 0 ) {
+                strncpy(projName, "Polar_Stereographic_", 21);
+                oSRS.SetProjection(SRS_PT_POLAR_STEREOGRAPHIC);
+                oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+            } else {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                     "Unknown projection.");
+            }
+        }
+
+        strcat(projName, target);
```

Comment:
risk of overflow

Reply:
Fixed in 41699ae24b983381a1a63715fbbae71c442059d8